src/poetry/core/packages/dependency.py

```diff
@@ -412,7 +407,7 @@ def _create_nested_marker(
                 if not constraint.include_min:
                     op = ">"
 
-                version = constraint.min.text
+                version: Version | str = constraint.min.text
```

Comment:
or I think this can always be a string (without needing annotation) if you set `version = constraint.max.text` a few lines later